MediaBrowser.Api/Playback/Hls/DynamicHlsService.cs

```diff
@@ -722,22 +722,155 @@ private bool EnableAdaptiveBitrateStreaming(StreamState state, bool isLiveStream
             //return state.VideoRequest.VideoBitRate.HasValue;
         }
 
-        private void AppendPlaylist(StringBuilder builder, StreamState state, string url, int bitrate, string subtitleGroup)
+        private string GetPlaylistAudioCodecs(StreamState state)
+        {
+            if (string.Equals(state.ActualOutputAudioCodec, "aac", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string profile = state.GetRequestedProfiles("aac").FirstOrDefault();
+                return HLSCodecStringFactory.GetAACString(profile);
+            }
+            else if (string.Equals(state.ActualOutputAudioCodec, "mp3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return HLSCodecStringFactory.GetMp3String();
+            }
+            else if (string.Equals(state.ActualOutputAudioCodec, "ac3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return HLSCodecStringFactory.GetAC3String();
+            }
+            else if (string.Equals(state.ActualOutputAudioCodec, "eac3", StringComparison.InvariantCultureIgnoreCase))
+            {
+                return HLSCodecStringFactory.GetEAC3String();
+            }
+            return string.Empty;
+        }
+
+        private string GetPlaylistVideoCodecs(StreamState state)
         {
-            var header = "#EXT-X-STREAM-INF:BANDWIDTH=" + bitrate.ToString(CultureInfo.InvariantCulture) + ",AVERAGE-BANDWIDTH=" + bitrate.ToString(CultureInfo.InvariantCulture);
+            int level = Convert.ToInt32(state.GetRequestedLevel(state.ActualOutputVideoCodec));
 
-            // tvos wants resolution, codecs, framerate
-            //if (state.TargetFramerate.HasValue)
-            //{
-            //    header += string.Format(",FRAME-RATE=\"{0}\"", state.TargetFramerate.Value.ToString(CultureInfo.InvariantCulture));
-            //}
+            if (string.Equals(state.ActualOutputVideoCodec, "h264", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string profile = state.GetRequestedProfiles("h264").FirstOrDefault();
+                return HLSCodecStringFactory.GetH264String(profile, level);
+            }
+            else if (string.Equals(state.ActualOutputVideoCodec, "h265", StringComparison.InvariantCultureIgnoreCase)
+                    || string.Equals(state.ActualOutputVideoCodec.ToLower() == "hevc", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
That doesn't look right

Reply:
Oh good catch, I must have went through too quickly when changing them all to string.Equals